lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -252,23 +294,39 @@ def find_matches(self, requirements):
                         raise ValueError(version_err) from ex
 
             return [
-                Candidate(fqcn, version, _none_src_server, first_req.type)
+                Candidate(fqcn, version, _none_src_server, first_req.type, None)
                 for version, _none_src_server in coll_versions
             ]
 
-        latest_matches = sorted(
-            {
-                candidate for candidate in (
-                    Candidate(fqcn, version, src_server, 'galaxy')
-                    for version, src_server in coll_versions
-                )
-                if all(self.is_satisfied_by(requirement, candidate) for requirement in requirements)
+        latest_matches = []
+        signatures = []
+        for version, src_server in coll_versions:
+            tmp_candidate = Candidate(fqcn, version, src_server, 'galaxy', None)
+
+            unsatisfied = False
+            if self._include_signatures:
+                signatures = src_server.get_collection_signatures(first_req.namespace, first_req.name, version)
+            for requirement in requirements:
+
+                unsatisfied |= not self.is_satisfied_by(requirement, tmp_candidate)
                 # FIXME
-                # if all(self.is_satisfied_by(requirement, candidate) and (
-                #     requirement.src is None or  # if this is true for some candidates but not all it will break key param - Nonetype can't be compared to str
-                #     requirement.src == candidate.src
-                # ))
-            },
+                # unsatisfied |= not self.is_satisfied_by(requirement, tmp_candidate) or not (
+                #    requirement.src is None or  # if this is true for some candidates but not all it will break key param - Nonetype can't be compared to str
+                #    or requirement.src == candidate.src
+                # )
+
+                if not self._include_signatures:
+                    continue
+                signatures.extend(
+                    [get_signature_from_source(url) for url in requirement.signature_sources or []]
```

Comment:
Seems like this `get_signature_from_source()` activity may belong to the artifact manager.